pkg/roachpb/metadata.go

```diff
@@ -946,7 +957,50 @@ func (h *GCHint) ForwardLatestRangeDeleteTimestamp(ts hlc.Timestamp) bool {
 	return false
 }
 
-// ResetLatestRangeDeleteTimestamp resets delete range timestamp.
-func (h *GCHint) ResetLatestRangeDeleteTimestamp() {
-	h.LatestRangeDeleteTimestamp = hlc.Timestamp{}
+// AddDeleteTimestamp adds the given deletion timestamp into the hint. Returns
+// true iff the hint was updated.
+func (h *GCHint) AddDeleteTimestamp(ts hlc.Timestamp) bool {
+	if ts.IsEmpty() {
+		return false
+	}
+	if h.MinDeleteTimestamp.IsEmpty() {
+		h.MinDeleteTimestamp = ts
+	} else if cmp := h.MinDeleteTimestamp.Compare(ts); cmp > 0 {
+		h.MinDeleteTimestamp = ts
```

Comment:
I think this needs to move the old MinDeleteTimestamp to MaxTimestamp if it's empty, otherwise we won't GC it eagerly.

Reply:
Good catch. Done this + exhaustive test that would have caught it.